tests/acceptance/users_frontend.feature

```diff
@@ -0,0 +1,46 @@
+Feature: Users Frontend
+  In order to manage users account in the web
+  As a user
+  I need to check user login and registration in joomla! CMS
+
+  Background:
+    Given I see the joomla! Home page
+
+  Scenario: Create user from frontend (index.php?option=com_users)
+    Given I click on the link "Create an account"
+    And I create a user with fields Name "patel", Username "patel", Password "patel" and Email "patel@gmail.com"
+    When I press the "Register"
+    Then I see "Could not instantiate mail function." message
+    And user is created
```

Comment:
This scenario should be

```Gherkin
Scenario: Create user from frontend (index.php?option=com_users)
    Given that user registration is enabled
    And there is no user with Username "patel" or Email "patel@gmail.com"
    When I click on the link "Create an account"
    And I create a user with fields Name "patel", Username "patel", Password "patel" and Email "patel@gmail.com"
    And I press the "Register"
    Then I see "Could not instantiate mail function." message
    And user is created
```

  - `I click on the link "Create an account"` is not a pre-condition, it is already part of your action, so it belongs to the `When` part.
  - `I see "Could not instantiate mail function." message` surely is not the expected message under the given pre-conditions. If you really mean this, add `And mail is not configured correctly` to the `Given` section. Otherwise, mail is assumed to be working as expected, and the message should contain `Your account has been created`.